framework/src/main/java/org/checkerframework/common/basetype/BaseTypeChecker.java

```diff
@@ -584,6 +586,22 @@ private void printStoredMessages(CompilationUnitTree unit) {
         if (messageStore != null) {
             for (CheckerMessage msg : messageStore) {
                 super.printOrStoreMessage(msg.kind, msg.message, msg.source, unit);
+                printStackTrace(msg.trace);
+            }
+        }
+    }
+
+    /**
+     * Output the given stack trace if the "dumpOnErrors" option is enabled.
+     *
+     * @param trace stack trace when the checker encountered a warning/error
+     */
+    private void printStackTrace(StackTraceElement[] trace) {
+        boolean dumpOnErrors = getBooleanOption("dumpOnErrors", false);
+        if (dumpOnErrors) {
+            for (StackTraceElement elem : trace) {
+                String msg = "\tat " + elem;
+                message(Diagnostic.Kind.NOTE, msg);
```

Comment:
We don't want individual messages for each element. Build up one string, including line breaks, and issue one NOTE.